src/app/pages/Movie/model/MovieStore.tsx

```diff
@@ -0,0 +1,95 @@
+import {
+  action,
+  computed,
+  makeObservable,
+  reaction,
+  runInAction,
+  type IReactionDisposer,
+} from 'mobx';
+
+import { getMovie, getMoviesByGenres } from '@/services/movies';
+import { LoadingStageModel } from '@/store/models/LoadingStageModel';
+import { ValueModel } from '@/store/models/ValueModel';
+import type { IMovie, IMovieShort } from '@/types/movies';
+import type { ILocalStore } from '@/types/store';
+
+export class MovieStore implements ILocalStore {
+  private _reactionDisposer?: IReactionDisposer;
+  private readonly _movie = new ValueModel<IMovie | null>(null);
+  private readonly _similarMovies = new ValueModel<IMovieShort[]>([]);
+
+  readonly loadingStage = new LoadingStageModel();
+  readonly similarLoadingStage = new LoadingStageModel();
+  private _movieId: number | null = null;
+
+  constructor(movieId: number) {
+    this._movieId = movieId;
+
+    makeObservable(this, {
+      movie: computed,
+      similarMovies: computed,
+      loadMovie: action.bound,
+      loadSimilarMovies: action.bound,
+    });
+
+    this._reactionDisposer = reaction(
+      () => [this._movieId],
+      () => {
+        this.loadMovie(movieId);
+      },
+      { fireImmediately: true }
+    );
+  }
+
+  get movie(): IMovie | null {
+    return this._movie.value;
+  }
+
+  get similarMovies(): IMovieShort[] {
+    return this._similarMovies.value;
+  }
+
+  async loadMovie(id: number) {
+    this.loadingStage.loading();
+    try {
+      const fetchedMovie = await getMovie(id);
+      runInAction(() => {
+        this._movie.change(fetchedMovie);
+        this.loadingStage.success();
+      });
+
+      this.loadSimilarMovies();
+    } catch {
+      runInAction(() => {
+        this.loadingStage.error();
+      });
```

Comment:
```suggestion
        this.loadingStage.error();
```

runInAction нужен, чтобы либо схлопнуть в один перерендер действия, которые изменяют observable поля, либо чтобы не оборачивать в action сам метод (вместо этого можно обернуть только конкретно место, где изменяется значения). Тут у тебя не подходит ни по один из этих случаев. Метод error уже обернут в action, второй раз не надо, после запроса в этой ветке только 1 действие, нет нужны ни с чем схлопывать

Reply:
Поправь, пожалуйста и в других местах